src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py

```diff
@@ -263,13 +263,22 @@ def _before_record_request(request):
                              '/graph.windows.net/{}/'.format(MOCKED_TENANT_ID), request.uri)
         request.uri = re.sub('/sig=([^/]+)&', '/sig=0000&', request.uri)
         request.uri = _scrub_deployment_name(request.uri)
-        # remove randomized portion from resource group name
-        resource_group_name = re.search('resource[Gg]roups/([^/]+_)/', request.uri)
+        # remove randomized portion from resource group name that is part of the URI
```

Comment:
if i understand the logic overall correctly that each test will produce one random resource group name, then to get things simpler, I would suggest we remember the random piece, and then do plain string replacement (with empty string) for both uri and body. We can make the random portion a little longer (to 6 characters)so to avoid conflicts.